### ISSUE TITLE
How to resolve servaldna does not exist error

### ISSUE THREAD

**Issue by error404-personnotfound** [servalproject/serval_chat#3]
Error:(3, 35) error: package org.servalproject.servaldna does not exist
Error:(15, 10) error: cannot find symbol class AbstractJsonList
where T,E are type-variables:
T extends Object declared in class AbstractFutureList
E extends Exception declared in class AbstractFutureList

![image](https://user-images.githubusercontent.com/30428236/32212243-0b3dc4be-be3c-11e7-842c-5fdce9e062e0.png)

**Comment by lakeman** [comment]
$ git submodule init
    $ git submodule update

Serval Chat doesn't have any getting started documentation yet, and we don't do any development on windows. 
A number of steps of the build process may need to be adjusted to support compiling libsodium and executing the git commands we are using to define the version code & name.

**Comment by error404-personnotfound** [comment]
Hey Jerry thanks for your support. As said by you I shifted from windows to
linux environment. Even then I am receiving the same error.
Can you please guide me as to when and where these commands need to be
entered

git submodule init

$ git submodule update

**Comment by gardners** [comment]
You need to run those commands before trying to compile.  They checkout
source code from other repositories that are required to compile the app.

Paul.

**Comment by error404-personnotfound** [comment]
This is the error which i get while configuring serval-dna

sqlite-amalgamation-3140200/sqlite3.c:1:9: error: unknown option after
‘#pragma GCC diagnostic’ kind [-Werror=pragmas]

Any way in which this can be resolved.

**Comment by error404-personnotfound** [comment]
rc/serval-dna$ make
SERVALD CC jni_common.c
SERVALD CC jni_commandline.c
SERVALD CC jni_instance.c
SERVALD CC jni_server.c
SERVALD CC sqlite-amalgamation-3140200/sqlite3.c
sqlite-amalgamation-3140200/sqlite3.c:1:9: error: unknown option after
‘#pragma GCC diagnostic’ kind [-Werror=pragmas]
 #pragma GCC diagnostic ignored "-Wimplicit-fallthrough"
         ^
cc1: all warnings being treated as errors
Makefile:254: recipe for target 'objs_servald/sqlite3.o' failed
make: *** [objs_servald/sqlite3.o] Error 1

**Comment by lakeman** [comment]
If you are having trouble building serval-dna, you should post that issue to serval-dna.

However, you shouldn't need to build serval-dna yourself to get started with Serval Chat, unless you need to produce a manifest for the apk you are building.

**Comment by error404-personnotfound** [comment]
How do I go about making an apk, if possible, from the serval chat code
when while making the build it gives an error that
Error:(6, 35) error: package org.servalproject.servaldna does not exist.

which forced me to configure serval-dna first.

Thanks for your continued support in this project.

**Comment by lakeman** [comment]
No, it is telling you that the java folder in the serval-dna submodule could not be found. Hence the above submodule commands.

**Comment by error404-personnotfound** [comment]
:app:libsodiumBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:libsodiumBuild'.
> Process 'command 'sh'' finished with non-zero exit value 127

This is the error while running ./gradlew assembleDebug


Another error which occurs during git update

~/serval_chat$ git submodule update
fatal: reference is not a tree: b6a9aafc13fb2477e28146af8bfb9b6be8b8e9a8
Unable to checkout 'b6a9aafc13fb2477e28146af8bfb9b6be8b8e9a8' in submodule
path 'app/src/main/jni/libsodium'


How can we resolve this error?

**Comment by lakeman** [comment]
Sigh, libsodium call it a stable branch, but they keep rebasing it.

**Comment by lakeman** [comment]
I've created and pushed a tag for each of the commit's we have used in serval chat's history to our own fork of libsodium.

So something like this should sort everything out;

    $ cd app/src/main/jni/libsodium
    $ git remote add serval https://github.com/servalproject/libsodium.git
    $ git fetch --tags serval
    $ cd ..
    $ git submodule update
